src/x/net/http/errors.go

```diff
@@ -92,25 +92,30 @@ func WriteError(w http.ResponseWriter, err error, opts ...WriteErrorOption) {
 		fn(&o)
 	}
 
+	statusCode := getStatusCode(err)
+	if o.response == nil {
+		w.Header().Set(HeaderContentType, ContentTypeJSON)
+		w.WriteHeader(statusCode)
+		json.NewEncoder(w).Encode(ErrorResponse{Error: err.Error()})
+	} else {
+		w.WriteHeader(statusCode)
+		w.Write(o.response)
+	}
+}
+
+func getStatusCode(err error) int {
 	switch v := err.(type) {
 	case Error:
-		w.WriteHeader(v.Code())
+		return v.Code()
 	case error:
 		if xerrors.IsInvalidParams(v) {
-			w.WriteHeader(http.StatusBadRequest)
+			return http.StatusBadRequest
 		} else if errors.Is(err, context.DeadlineExceeded) {
-			w.WriteHeader(http.StatusGatewayTimeout)
+			return http.StatusGatewayTimeout
 		} else {
-			w.WriteHeader(http.StatusInternalServerError)
+			return http.StatusInternalServerError
```

Comment:
Nit: I guess you can avoid both this `else` and the `default` case and just return the `StatusInternalServerError` at the end of the function.

Reply:
+1

Reply:
https://github.com/m3db/m3/pull/2917/commits/f8581442f5cdb07c43cecbfb3c785f4411f7634f#diff-9cc154bc6f46e06b0fa0f35fa2525acfc1a8582757d08940ba8f3f6045008240L115-R117